raiden-network/light-client
# dApp: refactoring transfer screen and fixes amount bug when no channels are open

Fixes #2031 
Fixes #1825 

**Short description**
This PR turned out a bit bigger because in the process of fixing the amount bug I also took the freedom to refactor the `Transfer` component which had become more complex than necessary throughout the iterations. In short, the field with menus and balance has been refactored to a separate component and also the input fields for address and amount are now in a separate component. There is where the scope of this refactoring ends, together with having adjusted the tests.

In the process, I created a separate issue for further refactoring of the deposit functionality #2074 

`No open channels` message displayed and the deposit button disabled.
<img width="653" alt="Screenshot 2020-08-10 at 13 35 48" src="https://user-images.githubusercontent.com/43838780/89778812-6fcdba00-db0e-11ea-8406-d326a3f1c3b7.png">

As a bonus, error messages no longer make the layout "jump".
<img width="613" alt="Screenshot 2020-08-10 at 11 53 58" src="https://user-images.githubusercontent.com/43838780/89771184-3726e400-db00-11ea-9b7b-54eb32268d27.png">

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Connect to the dApp and navigate to the transfer screen.
2. Click to access the channels menu and close a channel.
3. Navigate back to the transfer screen and "No open channels" should be displayed together with a disabled deposit button.
